exposed-core/src/main/kotlin/org/jetbrains/exposed/dao/id/EntityID.kt

```diff
@@ -32,6 +32,9 @@ open class EntityID<T : Comparable<T>> protected constructor(val table: IdTable<
     /** Performs steps when the internal [_value] is accessed without first being initialized. */
     protected open fun invokeOnNoValue() {}
 
+    /** Returns `true` if all component columns have been initialized with a value other than `null`. */
+    fun isNotInitialized(): Boolean = _value == null || (_value as? CompositeID)?.values?.any { it.value == null } == true
```

Comment:
Not sure about that, but is it correct that  `CompositeID` is uninitialized only if the user puts their null by itself, because by default `CompositeID::values` is empty, so there are no `null` values there?

`CompositeID {}` - Initialized
`CompositeId { it[Publishers.pubId] = null }` - Uninitialized

I tried to replace it with `fun isNotInitialized(): Boolean = _value == null ` to check what will fail, but the test are green yet.

It looks more reasonable if all the entity id parts from the table predefined as `null`s inside the `CompositeId`. But it would require to define id creation function on the table like `CompositeIdTable::createId(body)`. 

From one side it would protect user from partially created IDs (it would be possible to preset `null`s for parts, and check that after executen `body` there are no `null` values), also it would be possible to check that there are values only for the current table.

But it would make it less flexible and require always refer to particular table, so it's not the thing I'd actualy suggest.

This message is more about `isNotInitialized` check.

Reply:
I thought second part of the condition would cover, for example, optional references that store null for component columns, but new tests prove me wrong. So it seems it's only for manual nulls.

Checking to see if there are any other reasons to keep this, otherwise will scrap this function entirely before merging.